spec-tests/src/test/java/org/candlepin/spec/pools/OneSubPoolPerStackSpecTest.java

```diff
@@ -0,0 +1,805 @@
+    }
+
+    private List<PoolDTO> getHostPools(TestData data, List<String> contracts) {
+        return data.getHostClient().pools().listPoolsByConsumer(data.getHost().getUuid()).stream()
+            .filter(item -> contracts.contains(item.getContractNumber()))
+            .sorted(Comparator.comparing(item -> contracts.indexOf(item.getContractNumber())))
+            .collect(Collectors.toList());
+    }
+
+    private EntitlementDTO bindHostPool(TestData data, List<PoolDTO> pools, String contract) {
+        PoolDTO pool = pools.stream()
+            .filter(x -> contract.equals(x.getContractNumber()))
+            .collect(Collectors.toList()).get(0);
+        return ApiClient.MAPPER.convertValue(
+            data.getHostClient().consumers().bindPool(data.getHost().getUuid(), pool.getId(), 1).get(0),
+            EntitlementDTO.class);
+    }
+
+    private PoolDTO getSubPool(TestData data) {
+        return data.getGuestClient().pools().listPoolsByConsumer(data.getGuest().getUuid())
+            .stream()
+            .filter(x -> data.getStackId1().equals(x.getSourceStackId()))
+            .collect(Collectors.toList()).get(0);
+    }
+
+    private static class TestData {
+        OwnerDTO owner;
+        ApiClient userClient;
+        String guestVirtUuid;
+        ConsumerDTO guest;
+        ApiClient guestClient;
+        ConsumerDTO host;
+        ApiClient hostClient;
+        String stackId1;
+        String stackId2;
+
+        public TestData() {
+            owner = ownerApi.createOwner(Owners.random());
+            userClient = ApiClients.basic(UserUtil.createUser(client, owner));
+
+            guestVirtUuid = StringUtil.random("uuid");
+            guest = userClient.consumers().createConsumer(Consumers.random(owner)
+                .facts(Map.of("virt.is_guest", "true", "virt.uuid", guestVirtUuid)));
+            guestClient = ApiClients.ssl(guest);
+            host = userClient.consumers().createConsumer(Consumers.random(owner)
+                .guestIds(List.of(new GuestIdDTO().guestId(guestVirtUuid))));
+            hostClient = ApiClients.ssl(host);
+
+            stackId1 = StringUtil.random("stack");
+            stackId2 = StringUtil.random("stack");
+        }
+
+        public OwnerDTO getOwner() {
+            return owner;
+        }
+        public ApiClient getUserClient() {
+            return userClient;
+        }
+        public String getGuestVirtUuid() {
+            return guestVirtUuid;
+        }
+        public ConsumerDTO getGuest() {
+            return guest;
+        }
+        public ApiClient getGuestClient() {
+            return guestClient;
+        }
+        public ConsumerDTO getHost() {
+            return host;
+        }
+        public ApiClient getHostClient() {
+            return hostClient;
+        }
+        public String getStackId1() {
+            return stackId1;
+        }
+        public String getStackId2() {
+            return stackId2;
+        }
```

Comment:
Since the member variables have package-private modifiers, you don't need getters for the them. You should be able to just `TestData.owner`